src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -171,9 +313,13 @@ SyntaxToken getCloseQuote(Range openQuoteRange)
             var (expression, alignment) = getExpressionAndAlignment();
             var (format, closeBraceToken) = getFormatAndCloseBrace();
 
-            return SyntaxFactory.Interpolation(openBraceToken, expression, alignment, format, closeBraceToken);
+            var result = SyntaxFactory.Interpolation(openBraceToken, expression, alignment, format, closeBraceToken);
+#if DEBUG
```

Comment:
Any reason to `#ifdef` this assert?

Reply:
yes.  this would be an example of a very expensive assert.  it goes and fully 'ToStrings' the entire interpolation (which could be very large).  We wouldn't want those allocations during normal running.

Reply:
But `Debug.Assert` only exists in debug mode anyway (it's marked `[Conditional("DEBUG")]` https://source.dot.net/#System.Private.CoreLib/Debug.cs,81) and when the condition isn't met the whole call site is elided, so the `.ToFullString()` will never be called anyway. 

@jcouv to confirm the eliding behavior.

Reply:
ohhhhhhhhh.  i see what you mean.  yeah, we likely can remove that :)